### ISSUE TITLE
[Bug] Sankey桑基图 edgeLabel formatter中的value是data的value...

### ISSUE THREAD

**Issue by JohnRock0422** [apache/echarts#18325]
### Version

5.4.1

### Steps to Reproduce

```
option = {
  series: {
    type: 'sankey',
    layout: 'none',
    emphasis: {
      focus: 'adjacency'
    },
    data: [
      {name: 'data-0'},
      {name: 'data-1'},
      {name: 'data-2'}
    ],
    links: [
      {
        source: 'data-0', target: 'data-1',
        value: 01
      },
      {
        source: 'data-0', target: 'data-2',
        value: 02
      }
    ], 
    edgeLabel: {
      show: true, 
      formatter: '{b}'
    }
  }
};
```
There are 3 data, and 2 links
on `edgeLabel.formatter`
when it is showing the link[0], the formatter will show the value of data[0].
when it is showing the link[1], the formatter will show the value of data[1].
![sankey-link-data](https://user-images.githubusercontent.com/26728661/222659736-1cd33247-3908-489b-b4e9-d67a0a2f6c15.png)

### Current Behavior

when it is showing the link[0], the formatter will show the value of data[0].
when it is showing the link[1], the formatter will show the value of data[1].

### Expected Behavior

when it is showing the link[0], the formatter will show the value of link[0].
when it is showing the link[1], the formatter will show the value of link[1].

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

**Comment by helgasoft** [comment]
yes, looks like a bug. Probably related to _dataIndex_ mismatch on [this line](https://github.com/apache/echarts/blob/master/src/chart/sankey/SankeyView.ts#L227)  (?)
Workaround solution, by adding an extra (invisible) link, [Demo Code](https://echarts.apache.org/examples/en/editor.html?c=line-simple&theme=dark&code=PYBwLglsB2AEC8sDeBYAULWBnApgJwhywC5l1NMwBPEHUgciwENoBrHK-gGnIoBsmVYAFcwDaDBzdemHAFsQACyZYIJMhgqYAZsADGw9fSYATAFZM9OaHs4zYAXx6bMJpmCakA2vcxJoTHJ0sPRuHgC0AAz0Tr7IAUEMYUzhAIwxzlp-CcGh7ikATDH2ALqZ_BBs6j4uFEiwAPQNmJUAbmoQAEZ8OLA4AB5geEywfJWssGDAsNoQ_aNMnTh8WHGYWCJ4Vkn5UdyTTHgA5jhiIcnhReVZsK1MfMLBkddZY9A4AMrUPaRIoJYQajESKxWpaHAmE4AGUWy1-WEUwAA7qRtPdcA44qCbqgwRQNsItrkLtEuAdjqcdhF0i8tHcHk9UljaRobutNttzrtSeSTmc8hErmtbvdHqRIgUsaUyfYIdDYXxfnEEcjSENHjK8bo8HJ3GB8AwkJ0HPQ4k1tbqwPq8KQABQgACUCAAfMg8KdCXAQAA6ZLenIAanosAgJn68HoAZ9yQAktATAMANyY2opzAphyJoA)

![image](https://user-images.githubusercontent.com/13038071/222794963-6a85c007-f2ae-4532-b81b-f59e766805cf.png).
